sources/api/schnauzer/src/helpers.rs

```diff
@@ -969,77 +983,202 @@ pub fn kube_reserve_cpu(
+        .then(|| {
+            serde_json::from_value(hosts_value.clone()).context(
+                error::UnparseableTemplateValueSnafu {
+                    expected: "EtcHostsEntries",
+                    value: hosts_value.to_owned(),
+                    template: template_name.to_owned(),
+                },
+            )
+        })
+        .transpose()?;
+    trace!("Hosts from template: {:?}", hosts);
+
+    // If our hostname isn't resolveable, add it to the alias list.
+    if hostname.len() > 0 && !hostname_resolveable(hostname, hosts.as_ref()) {
+        results.push(hostname.to_owned());
+    }
+
+    // If hosts are specified and any overrides exist for loopback, add them.
+    if let Some(hosts) = hosts {
+        // If any static mappings in `settings.network.hosts` are for localhost, add them as well.
+        if let Some((_, aliases)) = hosts
+            .iter_merged()
+            .find(|(ip_address, _)| *ip_address == localhost_comparator)
+        {
+            // Downcast our hostnames into Strings and append to the results
+            let mut hostname_aliases: Vec<String> = aliases
+                .into_iter()
+                .map(|a| a.as_ref().to_string())
+                .collect();
+            results.append(&mut hostname_aliases);
+        }
     }
+
+    // Write out our localhost aliases.
+    let localhost_aliases = results.join(" ");
+    out.write(&localhost_aliases)
+        .context(error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
     Ok(())
 }
 
+/// This helper writes out /etc/hosts lines based on `network.settings.hosts`.
+///
+/// The map of <IpAddr => Vec<HostAlias>> is written as newline-delimited text lines.
+/// Any entries which reference localhost are ignored, as these are intended to be merged
+/// with the existing localhost entries via `localhost_aliases`.
+pub fn etc_hosts_entries(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name, if available.
+    trace!("Starting etc_hosts_entries helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly one (hosts overrides)
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 1)?;
+
+    // Get the resolved keys out of the template. value() returns a serde_json::Value
+    let hosts_value = helper
+        .param(0)
+        .map(|v| v.value())
+        .context(error::InternalSnafu {
+            msg: "Missing params after confirming that they exist.",
+        })?;
+    trace!("Hosts value from template: {}", hosts_value);
+
+    if hosts_value.is_null() {
+        // If hosts aren't set, just exit.
+        Ok(())
+    } else {
```

Comment:
It's a stylistic preference, but I would return here and unindent the "else" code.

```suggestion
    if hosts_value.is_null() {
        // If hosts aren't set, just exit.
       return  Ok(())
    }
```